elastic/kibana
# [Security Solution] [Sourcerer]  Add 'create runtime field' button to alerts and timeline browser fields

issue: https://github.com/elastic/security-team/issues/1723

## Summary

Add 'create runtime field' button to alerts and timeline browser fields 


### Browser field modal button
<img width="926" alt="Screenshot 2021-10-14 at 15 59 34" src="https://user-images.githubusercontent.com/1490444/137333099-90353801-8c1e-420e-b6d9-040365baad08.png">


### Timeline
<img width="1791" alt="Screenshot 2021-10-14 at 16 00 03" src="https://user-images.githubusercontent.com/1490444/137333090-3cb33f99-1b2c-4057-bedf-cbd46987c6aa.png">


### Alert page
<img width="1171" alt="Screenshot 2021-10-14 at 15 59 15" src="https://user-images.githubusercontent.com/1490444/137333102-f3df7932-7348-45ae-a7ac-b7b0e4b17f31.png">



## How to test it:
* Open Security solution alert page with data.
* Click on "fields" on the top of the alert table
* Click on create field button
* Set the field name and possibly the field value. Click on "save"
* The table should display a new column with the created field 


## Public API change
It includes one minor public API change of a typescript interface. However, the functionality doesn't change.
I typed `core.overlays.openFlyout` second parameter to accept `maskProps`. 
It was already working with this property because the options are spread into `EuiFlyout`.  
Here:
https://github.com/elastic/kibana/blob/a059b5df213355798b9c9c3caad8017dc3461c58/src/core/public/overlays/flyout/flyout_service.tsx#L139

### Checklist

Delete any items that are not applicable to this PR.
